File: C++/1-Two-Sum.cpp
#include<vector>
#include<unordered_map>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, size_t> tmp;
        for (size_t i = 0; i < nums.size(); ++i) {
            auto it = tmp.find(nums[i]);
            if (it == tmp.cend()) {
                tmp.emplace(target - nums[i], i);
            } else {
                return {it->second, i};
            }
        }
        return {};
    }
};

File: Python/1-Two-Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        tmp = dict()
        for i, n in enumerate(nums):
            diff = target - n
            if tmp.has_key(n):
                return list([tmp[n], i])
            else:
                tmp.update({diff: i})
        return list()
